pkg/dcmtime/common.go

```diff
@@ -0,0 +1,147 @@
+package dcmtime
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// isIncluded returns whether `check` is included in `limit`.
+//
+// Example: to test whether seconds should be included, you would:
+// isIncluded(PrecisionSeconds, [caller-passed-limit])
+func isIncluded(check PrecisionLevel, precision PrecisionLevel) bool {
+	return check <= precision
+}
+
+// truncateMilliseconds truncate nanosecond time.Time value to arbitrary precision.
+func truncateMilliseconds(
+	nanoSeconds int, precision PrecisionLevel,
+) (millisecondsStr string) {
+	milliseconds := nanoSeconds / 1000
+	millisecondsStr = fmt.Sprintf("%06d", milliseconds)
+	millisecondsStr = millisecondsStr[:6-(PrecisionFull-precision)]
+
+	return millisecondsStr
+}
+
+// Const for time.Time timezone. We don't want a timezone because we don't REALLY know
+// that a TM or DA value is UTC, so we don't want UTC to be made explicit.
+var zeroTimezone = time.FixedZone("", 0)
+
+// durationInfo holds parsed piece of DA, TM, or DT info.
+type durationInfo struct {
+	// The parsed int value
+	Value int
+	// Whether it was present in the string
+	Present bool
+	// Only used when parsing Fractal seconds: the level of precision used.
+	FractalPrecision PrecisionLevel
+}
+
+// extractDurationInfo extracts a piece of DA, TM, or DT info from a parsed regex.
```

Comment:
at first glance, it's a bit unclear to me what this function does. why does it need the entire subMatches array and an index, instead of just getting passed subMatches[i]?

Reply:
It's just a helper function, for each submatch I need to check that the index is in bounds, so I decided to handle that inside the function rather than every time the function was called.